momento-sdk/src/intTest/java/momento/sdk/CacheDataPlaneEagerConnectionTest.java

```diff
@@ -0,0 +1,39 @@
+package momento.sdk;
+
+import static momento.sdk.BaseTestClass.credentialProvider;
+import static momento.sdk.TestUtils.randomString;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import momento.sdk.config.Configurations;
+import momento.sdk.responses.cache.GetResponse;
+import momento.sdk.responses.cache.SetResponse;
+import org.junit.jupiter.api.Test;
+
+public class CacheDataPlaneEagerConnectionTest extends BaseTestClass {
+  private static final Duration DEFAULT_ITEM_TTL_SECONDS = Duration.ofSeconds(60);
+  private final String cacheName = System.getenv("TEST_CACHE_NAME");
+
+  @Test
+  void getReturnsHitAfterSet() {
+    CacheClient client =
+        CacheClient.create(
+            credentialProvider,
+            Configurations.Laptop.latest(),
+            DEFAULT_ITEM_TTL_SECONDS,
+            Duration.ofSeconds(10));
+    final String key = randomString("key");
+    final String value = randomString("value");
+
+    // Successful Set
+    final SetResponse setResponse = client.set(cacheName, key, value).join();
+    assertThat(setResponse).isInstanceOf(SetResponse.Success.class);
+    assertThat(((SetResponse.Success) setResponse).valueString()).isEqualTo(value);
+    assertThat(((SetResponse.Success) setResponse).valueByteArray()).isEqualTo(value.getBytes());
```

Comment:
this is more just a forward-thinking thing, but i wonder what it would take for us to upgrade the tests to use JDK 17 so that we can use JEP 394 in the tests? we obviously need to make sure the SDK itself can be used with JDKs at least back to 11, but for our own tests it sure would be nice to be able to use the modern patterns.

Reply:
I'm in favor of this if it's possible to have different language levels for the sdk and tests without having them in separate modules or something like that.

Reply:
For the interest of time, I have created an issue for this and will be happy to fast follow https://github.com/momentohq/client-sdk-java/issues/314. I agree with both of your sentiments though.